### PR TITLE
fix: sanitize usernames

### PR DESCRIPTION
**Pull request by sunnygleason**
Problem:

* icky usernames are failing the whole batch Keybase lookup

Solution:

* sanitize keybase usernames (imperfect but at least a first line of defense)

### CODE REVIEW

**Comment by sunnygleason** [comment]
fixes #452 (hopefully) 😂

File: api/api.js [hunk 1]
```diff
@@ -775,7 +776,12 @@ async function fetchValidatorAvatars(keybaseUsernames) {
   const avatarMap = new Map();
   let batch = keybaseUsernames.splice(0, MAX_KEYBASE_USER_LOOKUP);
   while (batch.length > 0) {
-    const usernames = batch.join(',');
+    // see: https://github.com/keybase/keybase-issues/issues/757#issuecomment-45229307
+    const usernames = flow(
+      uniq,
+      filter(x => x.match(/^[a-zA-Z0-9][a-zA-Z0-9_]{1,14}$/)),
```
Comment (jstarry) [inline api/api.js]: Nice regex! For completeness, we might as well filter out repeated underscores too

**Comment by sunnygleason** [comment]
> It would be nice to have a few tests as well!

I heartily agree! If possible, I would like to land this ahead of implementing testing infrastructure, and have created tracking ticket #515 for that work...

...And maybe I can even recruit you for some of those initial tests, since this is code I originally landed from someone you know... 😉 🍻 😂

File: api/api.js (hunk 1 above)
Comment (sunnygleason) [reply api/api.js]: 👍 awesome idea, will implement!

Comment (sunnygleason) [reply api/api.js]: done!

**Pull request merged**